packages/theme/src/cli/services/dev.ts

```diff
@@ -104,6 +106,33 @@ export async function dev(options: DevOptions) {
       renderWarning({headline: 'Failed to open the development server.', body: error.stack ?? error.message})
     })
   }
+
+  readline.emitKeypressEvents(process.stdin)
+  if (process.stdin.isTTY) {
+    process.stdin.setRawMode(true)
+  }
+
+  process.stdin.on('keypress', (_str, key) => {
+    if (key.ctrl && key.name === 'c') {
+      process.exit()
+    } else if (key.name === 't') {
+      openURL(`http://${host}:${port}`).catch((error) => {
+        renderWarning({headline: 'Failed to open localhost.', body: error.stack ?? error.message})
+      })
+    } else if (key.name === 'e') {
+      openURL(`https://${options.store}/admin/themes/${options.theme.id}/editor`).catch((error) => {
+        renderWarning({headline: 'Failed to open theme editor.', body: error.stack ?? error.message})
+      })
+    } else if (key.name === 'p') {
+      openURL(`https://${options.store}/?preview_theme_id=${options.theme.id}`).catch((error) => {
+        renderWarning({headline: 'Failed to open theme preview.', body: error.stack ?? error.message})
+      })
+    } else if (key.name === 'g') {
+      openURL(`http://${host}:${port}/gift_cards/${options.store}/preview`).catch((error) => {
+        renderWarning({headline: 'Failed to open gift card preview.', body: error.stack ?? error.message})
+      })
```

Comment:
These URLs are referenced elsewhere in this file for the links, correct? Could we extract them to variables or methods so there would only be one place to update them?

Same with all of the warning logs, they each have to do the same thing except for the actual `headline`. This could probably be a private method as well. I see we have at least one other `renderWarning` log on line 104 this file that could use the same private method as well